### PR TITLE
0.6.0 dev

### PR DESCRIPTION
**Pull request by YuanYuYuan**
Here are some upgrades on parts of throughput source codes from 0.5.0 to 0.6.0. To verify them, we can run 

```bash
cd throughput
cargo build --release --bin z_put_thr --bin z_sub_thr --bin zn_pub_thr --bin zn_sub_thr
```

### CODE REVIEW

**Comment by eddie9712** [comment]
I think there are two points can be confirmed:
* Different tests for zenoh/zenoh-net layer are needed or not.
* The format of argument "locator".

**Comment by william-wyliang** [comment]
> * Different tests for zenoh/zenoh-net layer are needed or not.

@eddie9712 I think some difference still remains in the merged API. You can see from the zenoh examples that [z_put.rs](https://github.com/william-wyliang/zenoh/blob/master/examples/examples/z_put.rs) and [z_pub.rs](https://github.com/william-wyliang/zenoh/blob/master/examples/examples/z_pub.rs) are still separated examples, in which the latter one was for zenoh-net. I think the use of [declare_expr](https://github.com/william-wyliang/zenoh/blob/4b4f106dbd6231f6817943bc9cdc41d1fd40ada2/examples/examples/z_pub.rs#L30) and [declare_publication](https://github.com/william-wyliang/zenoh/blob/4b4f106dbd6231f6817943bc9cdc41d1fd40ada2/examples/examples/z_pub.rs#L34) will be something worthy of note.

File: throughput/src/bin/z_put_thr.rs [hunk 1]
```diff
@@ -11,65 +11,76 @@
...
 #[derive(Debug, StructOpt)]
 #[structopt(name = "z_put_thr")]
 struct Opt {
-    #[structopt(short = "l", long = "locator")]
-    locator: String,
-    #[structopt(short = "m", long = "mode")]
+    #[structopt(
+        short,
+        long,
+        help = "locator(s), e.g. --locator tcp/127.0.0.1:7447 tcp/127.0.0.1:7448"
```
Comment (william-wyliang) [inline throughput/src/bin/z_put_thr.rs:33]: A suggestion here, as @eddie9712 has mentioned. Please discuss with other developers for the the syntax to specify multiple locators.

File: throughput/src/bin/z_put_thr.rs [hunk 2]
```diff
@@ -11,65 +11,76 @@
...
+    let Opt {
+        locator,
+        mode,
+        payload,
+        print,
+        config,
+    } = Opt::from_args();
+    let config = {
+        let mut config: Config = if let Some(path) = config {
+            json5::from_str(&async_std::fs::read_to_string(path).await.unwrap()).unwrap()
```
Comment (william-wyliang) [inline throughput/src/bin/z_put_thr.rs:63]: This is just a reminder. zenoh::Properties is replaced here. Please make sure the generated results are equivalent.

**Comment by eurc17** [comment]
LGTM

**Comment by william-wyliang** [comment]
Cargo.lock is auto-generated, right? Considering that, should we exclude this file from your commit? @YuanYuYuan

**Comment by YuanYuYuan** [comment]
Yes, I once had the same doubt. But Zenoh team may prefer including it into the git history (https://github.com/eclipse-zenoh/zenoh/blob/master/.gitignore#L5-L6).

**Review by william-wyliang — APPROVED** [review]
LGTM, with two reminding comments for a reference.
As for the question regarding Cargo.lock, according to the discussion in the comments, let's keep it in the PR at this moment.

File: throughput/src/bin/zn_sub_thr.rs [hunk 3]
```diff
@@ -11,76 +11,87 @@
...
+    #[structopt(short, long, help = "payload size (bytes)")]
     payload: usize,
-    #[structopt(short = "n", long = "name")]
+    #[structopt(short, long)]
     name: String,
-    #[structopt(short = "s", long = "scenario")]
+    #[structopt(short, long)]
     scenario: String,
-    #[structopt(long = "conf", parse(from_os_str))]
+    #[structopt(long = "conf", help = "configuration file (json5)")]
```
Comment (Mallets) [inline throughput/src/bin/zn_sub_thr.rs:44]: The configuration file could also be yaml

**Comment by Mallets** [comment]
> Yes, I once had the same doubt. But Zenoh team may prefer including it into the git history (https://github.com/eclipse-zenoh/zenoh/blob/master/.gitignore#L5-L6).

I would say that for the zenoh-perf repository, it is better to keep the `Cargo.lock` out. It is required in `zenoh` mainly (but not only) to deal with ABI compatibility. In caso of zenoh-perf, I believe it's good to not constrain it with a `Cargo.lock` that needs to be updated and maintained.

File: throughput/src/bin/z_put_thr.rs (hunk 2 above)
Comment (Mallets) [reply throughput/src/bin/z_put_thr.rs:63]: To create the config from file is enough to do: 
`Config::from_file(conf_file).unwrap()`
In this way, the config could also be in other formats than json5 (e.g. YAML)

**Pull request merged**